cli/aiconfig-editor/src/utils/promptUtils.ts

```diff
@@ -23,3 +24,69 @@ export function getPromptModelName(
   // Model must be specified as default if not specified for the Prompt
   return defaultConfigModelName!;
 }
+
+// TODO: Schemas should be statically defined with the model parsers and loaded alongside config, keyed on registered model names
+// Notes here: We could technically just refine the full JSON schema for the prompt since we currently allow model parsers to have full control, including:
+//   - defining metadata outside of model metadata --> e.g. remember_chat_context for openai parser
+//   - placing responsibility for proper handling/serialization of core config properties on the model parser (parameters, tags)
+// That would allow us to do some powerful things like:
+//   - share JSON schema for model parser between languages
+//   - validate serialization against it
+// Long-term, maybe we just have model parser core class handle serialization of parameters/tags and concrete implementations just
+// handle serialization of non-core metadata and model inference settings?
+// In any case, below we are defining 3 keys in the schema:
+//   - input: the prompt input, either string or more refined object including data / attachment types
+//   - model_settings: supported settings for the model, will be serialized either in top-level model metadata or prompt.metadata.model.settings
+//   - prompt_metadata: additional metadata for the prompt, EXCLUDES parameters, tags, model
+// A couple reasons for doing this vs refinining full JSON schema:
+//   - we may want to set constraints, like range of acceptable values, which aren't part of the prompt representation in JSON schema
+//   - we can focus on the important 3 params defined above, instead of requiring re-definining the full JSON schema
+// Should we define a JSON schema for PromptSchema type so we can safely serialize/deserialize them?
+export const PROMPT_SCHEMAS: Record<string, PromptSchema> = {
+  "gpt-3.5-turbo": OpenAIChatModelParserPromptSchema,
+  "gpt-4": OpenAIChatModelParserPromptSchema,
+};
+
+export type PromptInputSchema = {
+  type: "string" | "object";
+};
+
+export type ModelSettingsSchema = {
+  properties: Record<string, { [key: string]: any }>;
+  required?: string[];
+};
+
+export type PromptMetadataSchema = {
+  properties: Record<string, { [key: string]: any }>;
+  required?: string[];
+};
+
+export type PromptSchema = {
+  input: PromptInputSchema;
+  model_settings?: ModelSettingsSchema;
+  prompt_metadata?: PromptMetadataSchema;
```

Comment:
@saqadri I'm not really sure-- it seems like a lot of extra friction for anyone defining the Prompt schema for their parser. They'll need to include name and outputs (whose types should not be configurable for this purpose), then properly extend the metadata type (since parameters and tags are also not intended to be configured this way).

I think if we're defining these schemas as objects/dicts in the language of the parser then it's easier to do what you're suggesting. But if we need to change to have these schemas defined as a string (e.g. as a JSON schema) then the friction to do what you're suggesting is very high IMO